crates/pathfinder/src/rpc/api.rs

```diff
@@ -219,9 +222,10 @@ impl RpcApi {
                     .context("Fetch block from sequencer")
                     .map_err(internal_server_error)?;
 
+                let old_root = self.get_latest_block_root().await?;
                 let scope = requested_scope.unwrap_or_default();
 
-                return Ok(Block::from_sequencer_scoped(block, scope));
+                return Ok(Block::from_sequencer_scoped(block, old_root, scope));
```

Comment:
This is incorrect -- our latest block does not mean its the block before `pending` on sequencer.

Reply:
Correct, @kkovaacs mentioned it too.